assets/javascript/future-date.js

```diff
@@ -0,0 +1,13 @@
+function getCompareDate() {
+  var d = new Date(),
+      month = '' + (d.getMonth() + 1),
+      day = '' + d.getDate(),
+      year = d.getFullYear();
+  if (month.length < 2) month = '0' + month;
+  if (day.length < 2) day = '0' + day;
+  return [year, month, day].join('');
+}
+var elements = document.querySelectorAll('[future-date]');
+Array.prototype.forEach.call(elements, function(el, i){
+  if(el.getAttribute('future-date').split('-').join('') < getCompareDate()) el.remove();
+});
```

Comment:
I'd be reluctant to do any hand-parsing of dates. It's just too error-prone, when there are battle-tested utilities available. For this scenario, I think it can be done with built-in libraries, without even needing an extra dependency. 

This code filters out past events, so it could be adapted to filter out past events older than a month ago, or future events, or whatever criteria we want.
```
  const now = new Date().getTime(); // This gives a number, which is nice and comparable, rather than comparing individual elements
  Array.prototype.forEach.call(elements, function(el, i){
  const date = new Date(el.getAttribute('future-date')); // This might fiddling with, depending on what the format is of this attribute
        const isInThePast = date.getTime() < now;
        if(isInThePast) el.remove();
});
```

Reply:
(The code above *should* be a drop-in replacement for future-date.js, but I haven't actually tested it, and I'm a bit hazy on the `future-date` attribute and if it's trying to filter out past events or future events, so I might have swapped the logic.)